spec/controllers/dances_controller_spec.rb

```diff
@@ -28,12 +28,25 @@
       expect(assigns(:dance)).to eq(dance)
     end
 
-    it "refuses to show a private dance, instead redirecting back and giving a warning" do
-      dance = FactoryGirl.create(:dance, publish: false)
-      @request.env['HTTP_REFERER'] = '/'
-      get :show, params: {:id => dance.to_param}
-      expect(response).to redirect_to '/'
-      expect(flash[:notice]).to eq('this dance has not been published')
+    describe "without login" do
+      it "refuses to show a private dance, instead redirecting to login" do
+        dance = FactoryGirl.create(:dance, publish: false)
+        @request.env['HTTP_REFERER'] = '/'
+        get :show, params: {:id => dance.to_param}
+        expect(response).to redirect_to(new_user_session_path)
+        expect(flash[:notice]).to eq("that dance has not been published - maybe it's yours and you could see it if you logged in?")
+      end
+    end
+
+    describe 'with logged in user' do
+      login_user
+      it "refuses to show someone else's private dance" do
+        dance = FactoryGirl.create(:dance, publish: false)
+        @request.env['HTTP_REFERER'] = '/'
+        get :show, params: {:id => dance.to_param}
+        expect(response).to redirect_to(root_path)
+        expect(flash[:notice]).to eq("that dance has not been published so you can't see it")
+      end
```

Comment:
This is double-testing in light of the feature tests below. Can we remove these?